pkg/controllers/networking/pod_controller.go

```diff
@@ -773,6 +774,19 @@ func (r *PodReconciler) SetupWithManager(mgr ctrl.Manager) (err error) {
 			builder.WithPredicates(
 				&utils.IgnoreDeletePredicate{},
 				&predicate.ResourceVersionChangedPredicate{},
+				predicate.NewPredicateFuncs(func(obj client.Object) bool {
+					if r.PodSelector == nil {
+						return true
+					}
+
+					pod, ok := obj.(*corev1.Pod)
+					if !ok {
+						return false
+					}
+
+					// Only selected pods should be processed
```

Comment:
Except for Pod controller, should we also let  mutating & validating webhook be aware of the pod selector?

Reply:
yes, objectSelector field of `webhookConfigurations` should be set as same as this.